JSIL/AssemblyTranslator.cs

```diff
@@ -514,47 +528,47 @@ struct MethodToAnalyze {
 
             Action<int> writeAssembly = (i) => {
                 var assembly = assemblies[i];
+                bool stubbed = IsStubbed(assembly);
+
+                foreach (var pair in results) {
+                    long existingSize;
+                    foreach (var emmitterFactory in pair.EmitterGroup.MakeAssemblyEmitterFactory(this, assembly)) {
+                        var outputPath = FormatOutputFilename(emmitterFactory.AssemblyPathAndFilename);
+                        if (!Manifest.GetExistingSize(assembly, emmitterFactory.Id, out existingSize)) {
+                            using (var outputStream = new MemoryStream(DefaultStreamCapacity)) {
+                                var sourceMapBuilder = Configuration.BuildSourceMap.GetValueOrDefault() ? new SourceMapBuilder() : null;
+                                var context = MakeDecompilerContext(assembly.MainModule);
+
+                                try {
+                                    var tw = new StreamWriter(outputStream, Encoding.ASCII);
+                                    var formatter = new JavascriptFormatter(tw, sourceMapBuilder, this.TypeInfoProvider, Manifest, assembly, Configuration, stubbed);
+                                    var emitter = emmitterFactory.MakeAssemblyEmitter(formatter);
+                                    TranslateSingleAssemblyInternal(emitter, context, assembly, outputStream, sourceMapBuilder);
```

Comment:
Oh, this is super weird. Why are we translating the whole assembly once per emitter? Is there a reason why it has to work that way? It seems like it might be much better to do a single translation that calls out to all the emitters.

Otherwise, there's a risk that each individual translation could produce a slightly different result and the emitter outputs wouldn't match up.